dpe/src/x509.rs

```diff
@@ -2227,6 +2235,302 @@ impl CertWriter<'_> {
     }
 }
 
+enum CertificateFormat {
+    X509,
+    Csr,
+}
+
+enum CertificateType {
+    Leaf,
+    Exported,
+}
+
+pub(crate) struct CreateDpeCertArgs<'a> {
+    pub handle: &'a ContextHandle,
+    pub locality: u32,
+    pub cdi_label: &'a [u8],
+    pub key_label: &'a [u8],
+    pub context: &'a [u8],
```

Comment:
nit: Can you comment where each of these are used in derivation? Mainly that `context` is used for both (I think)